src/runtime/tests/type-checker-test.ts

```diff
@@ -426,6 +426,47 @@ describe('TypeChecker', () => {
     assert.deepStrictEqual(egressType.getEntitySchema(), concreteType.getEntitySchema());
   });
 
+  it('can resolve a type variable to its max type through an intermediary variable via a manifest', async () => {
+    const manifest = await Manifest.parse(`
+        particle OrderIngestion in '.OrderIngestion'
+          data: writes [Product {sku: Text, name: Text, price: Number}]
+
+        particle SkuRedactor in '.SkuRedactor'
+          input: reads [~a with {sku: Text}]
+          output: writes [~a]
+
+        particle Egress in '.Egress'
+          data: reads [~x with {sku: Text, *}]
+
+        recipe Shop
+          beforeRedaction: create
+          afterRedaction: create
+          OrderIngestion
+            data: beforeRedaction
+          SkuRedactor
+            input: beforeRedaction
+            output: afterRedaction
+          Egress 
+            data: afterRedaction
+    `);
+
+    const recipe = manifest.recipes[0];
+
+    const orderParticle = recipe.particles.find(p => p.name === 'OrderIngestion').spec;
+    const redactorParticle = recipe.particles.find(p => p.name === 'SkuRedactor').spec;
+    const egressParticle = recipe.particles.find(p => p.name === 'Egress').spec;
+
+    const concreteType = orderParticle.connections.find(c => c.name === 'data').type as CollectionType<EntityType>;
+    const egressType = egressParticle.connections.find(c => c.name === 'data').type as TypeVariable;
+
+    TypeChecker.processTypeList(null, [...orderParticle.connections, ...redactorParticle.connections]);
```

Comment:
For this test done call `TypeChecker.processTypeList`, just do `recipe.normalize()` and test that everything works end to end.

I think it's good that you have one test that's constructing typevars directly and calling `processTypeList`, and the other one that does stuff E2E (from manifest.parse to recipe.normalize)

Reply:
I would strongly recommend doing `recipe.normalize()` here instead of `TypeChecker.processTypeList`, so that you have a test that tests things entirely E2E. If you go one layer down to processTypeList you are prone to calling these APIs in a wrong way.